packages/commonwealth/server/routes/tokenBalance.ts

```diff
@@ -50,12 +59,50 @@ const tokenBalance = async (
   }
 
   try {
-    const balance = await tokenBalanceCache.fetchUserBalance(
-      chain.network,
-      chain_node_id,
-      req.body.address,
-      req.body.contract_address
-    );
+    let balance:
+      | string
+      | {
+          address: string;
+          balance: string;
+        }[];
+    if (req.body.all) {
+      const user_id = await models.Address.findOne({
+        where: {
+          chain: req.body.chain,
+          address: req.body.address,
+        },
+      });
+      const addresses: any = await models.Address.findAll({
+        attributes: [[fn('DISTINCT', col('address')), 'distinctAddress']],
+        where: {
+          chain: req.body.chain,
+          user_id: user_id.user_id,
+        },
+      });
+
+      if (addresses.length == 0) {
+        throw new AppError('No Addresses associated with user on this chain');
+      }
+
+      balance = await Promise.all(
+        addresses.map(async (address) => {
+          const addrBalance = await tokenBalanceCache.fetchUserBalance(
```

Comment:
Can we use `getBalancesForAddresses` instead of `Promise.all(fetchUserBalance())`?

Reply:
getBalancesForAddresses requires a balanceProvider as a param which we don't have here. And ultimately `getBalancesForAddresses` creates a `getBalance()` promis.all across all addresses provided. 
Implemented this way to avoid additional lookup of balance provider on the server side while basically calling the same logic